src/Raven.Studio/typescript/viewmodels/viewModelBase.ts

```diff
@@ -66,7 +66,21 @@ class viewModelBase {
         setTimeout(() => viewModelBase.showSplash(this.isAttached === false), 700);
         this.downloader.reset();
 
-        return this.databasesManager.activateBasedOnCurrentUrl();
+        const dbNameFromUrl = appUrl.getDatabaseNameFromUrl();
+        if (dbNameFromUrl) {
+            const dbAccessLevel = accessManager.default.getEffectiveDatabaseAccessLevel(dbNameFromUrl);
+            const requiredAccessForView = router.activeInstruction().config.requiredAccess;
+            const isViewAllowed = this.allowToAccessView(dbAccessLevel, requiredAccessForView);
+            
+            return this.databasesManager.activateBasedOnCurrentUrl(dbNameFromUrl, isViewAllowed);
+        } else {
+            return true;
```

Comment:
shouldn't we check here access as well?

Reply:
I'm not sure why we return true here w/o any checks. 

For views which are not database specific I assume we should check if we have required clearance or redirect to db lists.

Reply:
Access is now checked in the `canAccessView `method